pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -440,6 +450,43 @@ func (k *Kaniko) waitForJobCompletion(namespace string, jobName string, buildTim
 	return fmt.Errorf("Job has timed out. Job logs:\n%s", jobLogs)
 }
 
+func (k *Kaniko) resolveFailFast(namespace, jobName string, readinessTimoutSeconds *string) error {
+
+	// fail fast timeout is max(readinessTimeout, 5 minutes)
+	failFastTimeoutDuration, err := time.ParseDuration(*readinessTimoutSeconds)
+	if err != nil {
+		return errors.Wrap(err, "Failed to parse function readiness timeout")
+	}
+	if failFastTimeoutDuration < 5*time.Minute {
+		failFastTimeoutDuration = 5 * time.Minute
+	}
+	failFastTimeout := time.After(failFastTimeoutDuration)
+
+	// fail fast if job pod stuck in Pending or Unknown state
+	for {
+		select {
+		case <-failFastTimeout:
+			k.logger.WarnWith("Kaniko job was not completed in time",
+				"jobName", jobName,
+				"failFastTimeoutDuration", failFastTimeoutDuration)
+
+			return fmt.Errorf("Job was not completed in time, job name:\n%s", jobName)
+		default:
+			jobPod, err := k.getJobPod(jobName, namespace)
```

Comment:
in case of continue, add sleep of 5 second to avoid spamming k8s with get pod requests